components/style/properties/gecko.mako.rs

```diff
@@ -2959,19 +2964,36 @@ fn static_assert() {
             }
         %>
         ::values::generics::transform::TransformOperation::${name}${pattern} => {
-            bindings::Gecko_CSSValue_SetFunction(gecko_value, ${len(items) + 1});
+            % if has_optional:
+                let optional_present = ${items[-1] + str(len(items))}.is_some();
+                let len = if optional_present {
+                    ${len(items) + 1}
+                } else {
+                    ${len(items)}
+                };
+            % else:
+                let len = ${len(items) + 1};
+            % endif
+            bindings::Gecko_CSSValue_SetFunction(gecko_value, len);
             bindings::Gecko_CSSValue_SetKeyword(
                 bindings::Gecko_CSSValue_GetArrayItem(gecko_value, 0),
                 structs::nsCSSKeyword::eCSSKeyword_${keyword}
             );
             % for index, item in enumerate(items):
+                <% replaced_item = item.replace("optional_", "") %>
+                % if item.startswith("optional"):
```

Comment:
Ugh, this looks pretty hacky... :(

Reply:
There's no easy way to do it otherwise :(